src/lib/crypto/s2k.cpp

```diff
@@ -128,9 +133,64 @@ pgp_s2k_encode_iterations(size_t iterations)
      * For more info refer to rfc 4880 section 3.7.2.1.
      */
     for (uint16_t c = 0; c < 256; ++c) {
+        // This could be a binary search
         if (pgp_s2k_decode_iterations(c) >= iterations) {
             return c;
         }
     }
     return 255;
 }
+
+/// Should this function be elsewhere?
+static uint64_t get_timestamp_usec()
+   {
+   // Consider clock_gettime
+   struct timeval tv;
+   ::gettimeofday(&tv, NULL);
+   return (static_cast<uint64_t>(tv.tv_sec) * 1000000) + static_cast<uint64_t>(tv.tv_usec);
+   }
+
+size_t pgp_s2k_compute_iters(pgp_hash_alg_t alg, size_t desired_msec, size_t trial_msec)
+{
+   const uint64_t TRIAL_USEC = trial_msec * 1000;
+   const uint8_t MIN_ITERS = 96;
+   uint8_t buf[8192] = { 0 };
+   size_t bytes = 0;
+
+   if(desired_msec == 0)
+      desired_msec = DEFAULT_S2K_MSEC;
+   if(trial_msec == 0)
```

Comment:
Also, on the style side of things, [conditionals should always have braces](https://github.com/riboseinc/rnp/blob/master/doc/DEVELOPMENT-GUIDE.md#general-guidelines).

Reply:
Fixed, thanks